src/handler/updateVrm.ts

```diff
@@ -1,69 +1,40 @@
 
 export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
   logger.debug('Amend VRM Called');
-  try {
-    const isRequestInvalid: APIGatewayProxyResult | boolean = validateUpdateVrmRequest(event);
-    if (isRequestInvalid) {
-      return isRequestInvalid;
-    }
-    logger.debug('Request is Valid');
-    const systemNumber: string = decodeURIComponent(event.pathParameters?.systemNumber as string);
-    const createdTimestamp: string = decodeURIComponent(event.pathParameters?.createdTimestamp as string);
-    const { newVrm, isCherishedTransfer } = JSON.parse(event.body as string) as UpdateVrmRequestBody;
-    const currentRecord = await getBySystemNumberAndCreatedTimestamp(
-      systemNumber,
-      createdTimestamp,
-    );
-    const validateVrmRes = validateVrm(currentRecord, newVrm);
-    if (validateVrmRes) {
-      return validateVrmRes;
-    }
+  const isRequestInvalid: APIGatewayProxyResult | boolean = validateUpdateVrmRequest(event);
+  if (isRequestInvalid) {
+    return isRequestInvalid;
+  }
+  logger.debug('Request is Valid');
+  const systemNumber: string = decodeURIComponent(event.pathParameters?.systemNumber as string);
+  const createdTimestamp: string = decodeURIComponent(event.pathParameters?.createdTimestamp as string);
+  const { newVrm, isCherishedTransfer, newDonorVrm } = JSON.parse(event.body as string) as UpdateVrmRequestBody;
+  const currentRecord = await getBySystemNumberAndCreatedTimestamp(
+    systemNumber,
+    createdTimestamp,
+  );
+  const vrmsNotIncorrectFormat = validateVrm(currentRecord, newVrm);
+  if (vrmsNotIncorrectFormat) {
+    return vrmsNotIncorrectFormat;
+  }
 
-    const techRecords = await searchByCriteria(SearchCriteria.PRIMARYVRM, newVrm);
-    logger.debug('Tech record search returned: ', techRecords);
-    const filteredVrm = techRecords.filter((x) => x.primaryVrm === newVrm && x.techRecord_statusCode !== StatusCode.ARCHIVED);
-    if (filteredVrm.length) {
-      return addHttpHeaders({
-        statusCode: 400,
-        body: JSON.stringify(`Primary VRM ${newVrm} already exists`),
-      });
-    }
-    logger.debug('identifier has been validated');
-    const userDetails = getUserDetails(event.headers.Authorization ?? '');
-    const [recordToArchive, newRecord] = processPatchVrmRequest(currentRecord, userDetails, newVrm, isCherishedTransfer);
+  const userDetails = getUserDetails(event.headers.Authorization ?? '');
 
-    if (isCherishedTransfer) {
-      await updateVehicle(
-        [recordToArchive],
-        [newRecord],
-      );
-    } else {
-      await correctVrm(newRecord);
-    }
-    return addHttpHeaders({
-      statusCode: 200,
-      body: JSON.stringify(formatTechRecord(newRecord)),
-    });
-  } catch (error) {
-    console.log(error);
-    return addHttpHeaders({
-      statusCode: 500,
-      body: JSON.stringify(error),
-    });
+  if (isCherishedTransfer) {
```

Comment:
I'm not sure I like this new pattern. I'd like to keep the endpoints semantically the same, so we call a processor, that returns the value, we then call to the database, that returns, then we send a response. I don't want calls nested down if possible as I want top layer visibility if that makes sense?

Reply:
Yeah that's fine, I can change that so that the handler is doing the calls to the db and the processor just formats and returns the records that's no problem